kiosk/api/queries/getHero/index.ts

```diff
@@ -0,0 +1,108 @@
+import { fetchGraphQL } from '../..';
+import { HeroResponse, HeroResult, Image } from '../../../interfaces/hero';
+
+export const getHeroes = async (): Promise<{
+  heroes: HeroResult[];
+}> => {
+  let id: string = process.env.NEXT_PUBLIC_CMP_HERO_CONTENT_ID || '';
+  if (id === '') id = 'jnZOQnEVhUyWn1le6m5FyQ';
+  const heroQuery = `
+    query {
+      allM_Content_Advertisement(where:{id_eq:"${id}"}) {
+        results {
+          id
+          advertisement_Title
+          advertisement_Body
+          advertisement_Slogan
+          advertisement_Eyebrow
+          content_Name
+          advertisement_Logo{
+            results{
+               assetToPublicLink(first: 1) {
+                results {
+                  id
+                  relativeUrl
+                  versionHash
+                }
+              }
+            }
+          }
+          advertisement_Image: cmpContentToMasterLinkedAsset{
+            results{
+               assetToPublicLink(first: 1) {
+                results {
+                  id
+                  relativeUrl
+                  versionHash
+                }
+              }
+            }
+          }
+          advertisement_Background{
+            results{
+               assetToPublicLink(first: 1) {
+                results {
+                  id
+                  relativeUrl
+                  versionHash
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    `;
+
+  if (process.env.CI === 'true') {
+    return { heroes: [] as HeroResult[] };
+  }
+
+  const results: HeroResponse = (await fetchGraphQL(heroQuery)) as HeroResponse;
+  const heroes: HeroResult[] = results?.data.allM_Content_Advertisement.results;
+
+  return {
+    heroes: heroes,
+  };
+};
+
+export const getHero = async (): Promise<{ hero: HeroResult }> => {
+  const { heroes } = await getHeroes();
+  const dummyImage: Image = {
+    results: [
+      {
+        id: 'dummy',
+        fileName: 'dummy.jpg',
+        assetToPublicLink: {
+          results: [
+            {
+              id: 'dummy',
+              relativeUrl: 'b1a88e26f6a54a9ea64a5f759c5eea84',
+              versionHash: '5aea50fb',
+            },
+          ],
+        },
+      },
+    ],
+  };
+
+  if (heroes?.length > 0) {
+    return {
+      hero: heroes[0],
+    };
+  } else {
+    return {
+      hero: {
+        id: 'dummy',
```

Comment:
In which case is it necessary to return a dummy hero with dummy images?

Is it when the CMP environment variables are not set? In that case, I would refuse to start the app or expect it breaks. If we introduce a CMP dependency, it is expected the app is not working when the environment variables are not provided. In that case, I would remove these dummy fallbacks.

Reply:
I was getting this error on build, thats why I created it. Can you please look into it.
https://dev.azure.com/sitecoredst/Demo/_build/results?buildId=27499&view=logs&j=ef7440c6-a6d6-5157-e050-1a43bc10534f&t=f0850fbf-e2ab-521d-2b9a-29201b5f7565&l=33

Reply:
Building the TV/Kiosk apps in our CI pipeline is done only to catch TypeScript type validity errors (and maybe other errors too) before the code is built by Vercel. When we develop the apps and when Vercel builds the apps, environment variables are defined and valid. However, when our CI pipeline is building the apps, it does not have the environment variables. This currently forces us to add dirty workarounds like these fallback values in our code.

I think adding the CH environment variables and their values to our CI pipeline would allow us to remove these fallbacks.

Reply:
I created work item 6820 for this issue. We will address it separately.